addons/account_edi/models/account_edi_document.py

```diff
@@ -59,21 +68,31 @@ def _prepare_jobs(self):
                 continue
 
             key = (edi_format, edi_doc.state, doc_type, move.company_id)
+            if not to_process.get(key, None):
+                to_process[key] = self.env['account.edi.document']
+            to_process[key] |= edi_doc
+
+        # Order payments/invoice and create batches.
+        result = []
+        payments = []
+        for key, docs in to_process.items():
+            edi_format, edi_doc.state, doc_type, move.company_id = key
+            target = result if doc_type == 'invoice' else payments
             if edi_format._support_batching():
-                if not batches.get(key, None):
-                    batches[key] = self.env['account.edi.document']
-                batches[key] |= edi_doc
+                target.append((docs, doc_type))
             else:
-                to_process.append((key, edi_doc))
-        to_process.extend(batches.items())
-        return to_process
+                for doc in docs:
+                    target.append((docs, doc_type))
+        result.extend(payments)
+        return result
 
     @api.model
-    def _process_jobs(self, to_process):
+    def _process_job(self, documents, doc_type):
```

Comment:
you can't change the method signature in a stable version. I don't understand why you need to do that: to discuss

Reply:
I don't need to do that, it's just simpler. However, if this is not allowed, I will revert back.